src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs

```diff
@@ -155,17 +155,28 @@ private async Task ParseFileOperationAsync(PipeStream connection, Dictionary<str
 
                                 var shellOperationResult = new ShellOperationResult();
 
-                                using var shd = new ShellFolder(Path.GetDirectoryName(filePath));
-                                op.QueueNewItemOperation(shd, Path.GetFileName(filePath),
-                                    (string)message["fileop"] == "CreateFolder" ? FileAttributes.Directory : FileAttributes.Normal, template);
+                                if (!Extensions.IgnoreExceptions(() =>
+                                {
+                                    using var shd = new ShellFolder(Path.GetDirectoryName(filePath));
+                                    op.QueueNewItemOperation(shd, Path.GetFileName(filePath),
+                                        (string)message["fileop"] == "CreateFolder" ? FileAttributes.Directory : FileAttributes.Normal, template);
+                                }))
+                                {
+                                    shellOperationResult.Items.Add(new ShellOperationItemResult()
+                                    {
+                                        Succeeded = false,
+                                        Destination = filePath,
+                                        HRresult = (int)-1
+                                    });
+                                }
 
                                 var createTcs = new TaskCompletionSource<bool>();
                                 op.PostNewItem += (s, e) =>
                                 {
                                     shellOperationResult.Items.Add(new ShellOperationItemResult()
                                     {
                                         Succeeded = e.Result.Succeeded,
-                                        Destination = e.DestItem?.FileSystemPath,
+                                        Destination = e.DestItem != null ? (e.DestItem.IsFileSystem ? e.DestItem.FileSystemPath : e.DestItem.ParsingName) : null,
```

Comment:
Can some kind of helper method be written to avoid duplicated checks?

Reply:
Done! Looks a lot better now xD